precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/route/SceneRoute.kt

```diff
@@ -10,5 +11,28 @@ internal class SceneRoute(
     val deepLinks: List<String>,
     val navTransition: NavTransition?,
     val swipeProperties: SwipeProperties?,
-    override val content: @Composable (BackStackEntry) -> Unit,
+    override val content: @Composable AnimatedContentScope.(BackStackEntry) -> Unit,
 ) : ComposeRoute, ComposeSceneRoute
+
+@Deprecated(
+    message = """
+    Used as a backwards compatible for the old RouteBuilder APIs which do not expect the content to
+    be an extension function on AnimatedContentScope        
+    """,
+    level = DeprecationLevel.WARNING,
+)
+internal fun sceneRouteWithoutAnimatedContent(
```

Comment:
I wanted to do something like what androidx.navigation does here https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:navigation/navigation-compose/src/main/java/androidx/navigation/compose/NavGraphBuilder.kt;l=57-60?q=%22ComposeNavigator.Destination(provider%22 which calls this internal deprecated function here https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:navigation/navigation-compose/src/main/java/androidx/navigation/compose/ComposeNavigator.kt;l=106-110 but for them it worked because the API also changed in a meaningful way besides just adding the AnimatedContentScope.

In our case here if I add it as a secondary constructor then there's ambiguity between the two of them and it's impossible to differentiate between the two. So I opted for an internal deprecated function to exist solely for this purpose.